packages/utilities/src/scroll.ts

```diff
@@ -146,35 +146,32 @@ export function getScrollbarWidth(): number {
 }
 
 /**
- * Traverses up the DOM for the element with the data-is-scrollable=true attribute, or returns
- * document.body.
+ * Traverses up the DOM for the element with the data-is-scrollable=true attribute and parallelly
+ * checks if the overflow property of an element allows the element to scroll (while excluding
+ * the elements with absolute, static and fixed positioning) , or finally returns document.body.
  *
  * @public
  */
 export function findScrollableParent(startingElement: HTMLElement | null): HTMLElement | null {
   let el: HTMLElement | null = startingElement;
-
+  const overflowRegex = /(auto|scroll)/;
   // First do a quick scan for the scrollable attribute.
   while (el && el !== document.body) {
+    let style = getComputedStyle(el);
+    if (style.position === 'fixed') {
+      return document.body;
+    }
     if (el.getAttribute(DATA_IS_SCROLLABLE_ATTRIBUTE) === 'true') {
       return el;
-    }
-    el = el.parentElement;
-  }
-
-  // If we haven't found it, the use the slower method: compute styles to evaluate if overflow is set.
-  el = startingElement;
-
-  while (el && el !== document.body) {
-    if (el.getAttribute(DATA_IS_SCROLLABLE_ATTRIBUTE) !== 'false') {
-      const computedStyles = getComputedStyle(el);
-      let overflowY = computedStyles ? computedStyles.getPropertyValue('overflow-y') : '';
-
-      if (overflowY && (overflowY === 'scroll' || overflowY === 'auto')) {
+    } else {
+      if (style.position === 'absolute' || style.position === 'static') {
+        el = el.parentElement;
```

Comment:
I'm nervous that adding a style-based check may actually cause false-positives, where the system uses an element which is accidentally scrollable, rather than the element actually intended for scrolling. Just because a component has `position: absolute` does not mean it's the one we want to scroll. In fact, `position: absolute` may be used *within* scrollable elements without issue, so a child of such a element may not get the correct scrollable parent!